### PR TITLE
Fix crash when direct boot

### PR DESCRIPTION
**Pull request by AnatolyPristensky**
* [X] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.

## Description

Fix crash when direct boot is used. 

## Related PRs or issues

#1599
[AB#91315](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/91315)

### CODE REVIEW

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java [hunk 1]
```diff
@@ -102,7 +106,16 @@ private static SharedPreferencesTestData[] generateSharedPreferenceData() throws
 
         return testData;
     }
-
+    @Test
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java:109]: Empty space before and after this method

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java [hunk 2]
```diff
@@ -102,7 +106,16 @@ private static SharedPreferencesTestData[] generateSharedPreferenceData() throws
...
+    public void sharedPreferencesProtectedStorage() {
+        Boolean isProtectedStorage = context.isDeviceProtectedStorage();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java:112]: Condition in test function? Wouldn't it better to separate the testing on two test methods?

File: sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java [hunk 3]
```diff
@@ -102,7 +106,16 @@ private static SharedPreferencesTestData[] generateSharedPreferenceData() throws
...
+            assertEquals(isProtectedStorage, false);
+        }
+        else {
```
Comment (DmitriyKirakosyan) [inline sdk/appcenter/src/androidTest/java/com/microsoft/appcenter/utils/storage/SharedPreferencesManagerAndroidTest.java:115]: `else` on the same line with `}`

**Review by MatkovIvan — COMMENTED** [review]
Need to be tested manually before merge

**Comment by DmitriyKirakosyan** [comment]
@AnatolyPristensky I've tested your changes and found a problem. In this case, the updated SDK will create new preferences and thus the previously saved data will be lost.
Instead of writing lots of code implementing the migration from old preferences to the new ones, It would make sense to create the protected preferences only in the case of the exception. Let's create normal shared preferences as we did before and handle the error, and only create protected shared preferences in case we catch the error. This way we can avoid supporting the migration between preferences.

**Pull request closed without merging**